packages/common-server/src/parser.ts

```diff
@@ -301,16 +303,51 @@ export class SchemaParserV2 extends ParserBaseV2 {
     });
 
     const addConnections = (parent: SchemaProps) => {
-      _.forEach(parent.children, (ch) => {
-        const child = schemasDict[ch];
-        if (child) {
-          DNodeUtils.addChild(parent, child);
+      _.forEach(parent.children, (childId) => {
+        const child = schemasDict[childId];
 
-          addConnections(child);
+        if (child) {
+          if (child.parent === null) {
+            // Child does not have a parent pointers hence we can linkup the parent child as is.
+            DNodeUtils.addChild(parent, child);
+            addConnections(child);
+          } else {
+            // Child already contains a parent pointer hence we need to create a schema clone
+            // otherwise we would end up writing over the already existing parent pointer.
+            const childClone = _.cloneDeep(child);
+            // We use a dictionary hence we need to generate a brand new identifier for our clone.
+            childClone.id = `${childId}_${genUUID()}`;
+            schemasDict[childClone.id] = childClone;
+
+            // When it comes to schemas we often use the id as a matching pattern
+            // unless there is a pattern explicitly specified. Since we created a randomized
+            // identifier the id of our clone no longer can be used for pattern matching.
+            // Hence if there isn't a pattern we will set the pattern to match original id.
+            if (
+              childClone.data === undefined ||
+              childClone.data.pattern === undefined
+            ) {
+              if (childClone.data === undefined) {
```

Comment:
heads up that you can use `_.set(childClone, "data.pattern", childId)` to simplify

Reply:
Added _.set